stable-patch-format: stricter rule for subject line + minor cosmetic changes (#48)
canonical/kernel-docs: docs/reference/stable-patch-format.rst

starting with "[SRU]" followed by the release name against which the
patch is targeted.

The release name **must** be enclosed in "[ ]" brackets and **should** be
abbreviated using the first letter of the release name (e.g. "N" for
"Noble") in upper case. For example:

.. code-block:: none

   [SRU][N][PATCH 0/1] Fix error of resume on rtl8168fp

* If a patch is to be applied to multiple releases, a list of release names

>   must be provided, with "/" separating the uppercase letter representing each release. For example, when it targets Bionic and Focal:

Reviewer: Nitpick: the max line length here is inconsistent.

Author: Thanks for the review. Fixed in https://github.com/canonical/kernel-docs/pull/48/commits/803e692663cfc040bf24f56b305cb43afa190c43.

  .. code-block:: none

     [SRU][B/F][PATCH 1/1] KVM: fix overflow of zero page refcount with ksm running

  .. note::

     Historically, the rule has been somewhat flexible, and various styles
     have been permitted.  You may find examples of various styles (such as the
     ones below) in the `mailing list archive

>      <https://lists.ubuntu.com/archives/kernel-team/>`__:

Reviewer: ```suggestion
     <https://lists.ubuntu.com/archives/kernel-team/>`_:
```

Author: Thanks for the review. Fixed in https://github.com/canonical/kernel-docs/pull/48/commits/803e692663cfc040bf24f56b305cb43afa190c43.

     .. code-block:: none
        :linenos:

        [SRU][B,F][PATCH 1/1] KVM: fix overflow of zero page refcount with ksm running
        [SRU][B][F][PATCH 1/1] KVM: fix overflow of zero page refcount with ksm running
        [SRU][Bionic,Focal][PATCH 1/1] KVM: fix overflow of zero page refcount with ksm running
        [SRU][Bionic/Focal][PATCH 1/1] KVM: fix overflow of zero page refcount with ksm running
        [SRU][Bionic][Focal][PATCH 1/1] KVM: fix overflow of zero page refcount with ksm running

     Please adhere to the ``[B/F]`` style.

     * Use initial letter(s) in upper case
     * Separate each series with "/"

* If the patch has to be applied to a specific derivative for multiple
  releases, indicate the derivative after the release. For example:

  .. code-block:: none

>      [SRU][B:linux-kvm/F:linux-kvm] [PATCH 0/1] UBUNTU: [Config] kvm: Add support for modifying LDT

Reviewer: ```suggestion
     [SRU][B:linux-kvm/F:linux-kvm][PATCH 0/1] UBUNTU: [Config] kvm: Add support for modifying LDT
```

Nitpick: not sure if the PATCH needs a whitespace before it but i see most of the examples don't have it.

Author: Thanks for the review. Fixed in https://github.com/canonical/kernel-docs/pull/48/commits/803e692663cfc040bf24f56b305cb43afa190c43.

starting with "[SRU]" followed by the release name against which the
patch is targeted.

The release name **must** be enclosed in "[ ]" brackets and **should** be
abbreviated using the first letter of the release name (e.g. "N" for

> "Noble"). For example:

Reviewer: ```suggestion
"Noble") in upper case. For example:
```

.. code-block:: none

   [SRU][N][PATCH 0/1] Fix error of resume on rtl8168fp

* If a patch is to be applied to multiple releases, a list of release names

>   must be provided. For example, when it targets Bionic and Focal:

Reviewer: ```suggestion
  must be provided, with "/" separating the letter representing each series. For example, when it targets Bionic and Focal:
```

Reviewer: I know this is mentioned below but that is in a notebox which should supplement the main content and "theoretically" can be skipped.
I suggest just making it explicitly clear here.

  .. code-block:: none

     [SRU][B/F][PATCH 1/1] KVM: fix overflow of zero page refcount with ksm running

  .. note::

     Historically, the rule has been somewhat flexible, and various styles

>      have been permitted. You may find examples like the following in the
>      `mailing list archive
>      <https://lists.ubuntu.com/archives/kernel-team/>`__:

Reviewer: ```suggestion
     have been permitted.  You may find examples of various styles (such as the
     ones below) in the `mailing list archive
     <https://lists.ubuntu.com/archives/kernel-team/>`__:
```

Reviewer: "Such as the ones below" to make it clear they are all separate entries without using "OR" within the code block itself.
This is what it would look like:
![image](https://github.com/user-attachments/assets/d5555205-1f59-4989-b425-c465932d4ed1)

Author: Got it, thanks.

>      .. code-block:: none

Reviewer: ```suggestion
     .. code-block:: none
        :linenos:
```

        [SRU][B,F][PATCH 1/1] KVM: fix overflow of zero page refcount with ksm running

>         or

Reviewer: I don't think it's a good practice to mix in the "OR"s with the examples you're trying to demonstrate
I suggest adding line numbers instead to help with the description before the code block.

Author: Got it, thanks.

>         [SRU][B,F][PATCH 1/1] KVM: fix overflow of zero page refcount with ksm running
>         or
>         [SRU][B][F][PATCH 1/1] KVM: fix overflow of zero page refcount with ksm running
>         or
>         [SRU][Bionic,Focal][PATCH 1/1] KVM: fix overflow of zero page refcount with ksm running
>         or
>         [SRU][Bionic/Focal][PATCH 1/1] KVM: fix overflow of zero page refcount with ksm running
>         or
>         [SRU][Bionic][Focal][PATCH 1/1] KVM: fix overflow of zero page refcount with ksm running

Reviewer: ```suggestion
        [SRU][B,F][PATCH 1/1] KVM: fix overflow of zero page refcount with ksm running
        [SRU][B][F][PATCH 1/1] KVM: fix overflow of zero page refcount with ksm running
        [SRU][Bionic,Focal][PATCH 1/1] KVM: fix overflow of zero page refcount with ksm running
        [SRU][Bionic/Focal][PATCH 1/1] KVM: fix overflow of zero page refcount with ksm running
        [SRU][Bionic][Focal][PATCH 1/1] KVM: fix overflow of zero page refcount with ksm running
```

   Example:

   .. code-block:: none
      :emphasize-lines: 2-5
      :emphasize-lines: 2-4

>       Subject: [PATCH 1/1][SRU][B] UBUNTU: SAUCE: platform/x86: dell-uart-backlight: add get_display_mode command

Reviewer: Oh just realised the PATCH came before SRU and release name.
Good catch.

     .. code-block:: none

        (cherry picked from commit <sha1> <upstream repo name>)

>      .. note::

Reviewer: hi @lkpdn , the indentation here seems to imply that this note only applies to cheery-picked patches.
Is that the case?

Also I think there was a question about this <upstream repo name> and what should be indicated here.
Maybe this note needs to be mentioned directly instead.

Reviewer: For example

   Backported patches:
   - If the patch required changes (e.g. it did not apply cleanly), use
     "backported from commit <sha1> <upstream repo name>" between brackets "()". Omit the "<upstream repo name>" if the patch comes from the mainline tree. For example:

Author: You're right. Just my mistake. I'll fix it, thanks!
